crates/core/tedge/src/cli/certificate/renew.rs

```diff
@@ -0,0 +1,84 @@
+use super::error::CertError;
+use super::remove::RemoveCertCmd;
+use crate::command::Command;
+use crate::CreateCertCmd;
+use camino::Utf8PathBuf;
+use certificate::NewCertificateConfig;
+
+pub struct RenewCertCmd {
+    /// The path where the device certificate will be tored
+    pub id: String,
+    pub cert_path: Utf8PathBuf,
+    pub key_path: Utf8PathBuf,
+}
+
+impl Command for RenewCertCmd {
+    fn description(&self) -> String {
+        "Renew a self-signed certificate for the device.".into()
+    }
+
+    fn execute(&self) -> anyhow::Result<()> {
+        let config = NewCertificateConfig::default();
+        self.renew_test_certificate(&config)?;
+        eprintln!("Certificate was successfully renewed, for un-interrupted service, the certificate has to be uploaded to the cloud");
+        Ok(())
+    }
+}
+
+impl RenewCertCmd {
+    fn renew_test_certificate(&self, config: &NewCertificateConfig) -> Result<(), CertError> {
+        // Remove existing certificate
+        let rm_cmd = RemoveCertCmd {
```

Comment:
Before removing the certificate, I would add a check that the certificate exists and is actually self-signed.

Reply:
checking if the certificate is `self-signed`